openlibrary/catalog/add_book/__init__.py

```diff
@@ -678,6 +684,70 @@ def load_data(rec, account_key=None):
     return reply
 
 
+def overwrite_promise_item_using_marc_record(
+    matched_edition: Edition, incoming_rec: dict
+) -> tuple[Edition, bool]:
+    """
+    This (almost) entirely writes over a matched edition with data from
+    incoming record created by POSTing to /api/items/ia.
+
+    This is for when a matched edition is known to have data that is
+    generally less reliable than the incoming record (e.g. because
+    the incoming record is based on MARC data and the matched edition is a
+    revision 1 promise item.)
+    """
+    need_edition_save = False
+    incoming_edition_dict = build_query(incoming_rec)
+
+    # Copy the higher quality data from incoming_rec into the matched edition,
+    # but only do so if the information is actually different so we know
+    # whether to save. Use a dictionary representation of `matched_edition` to
+    # get the data representation as close as possible for easy copying, though
+    # some cases, e.g. `authors`, must be handled case by case because the
+    # representations are still different.
+    matched_edition_dict: dict = matched_edition.dict()
+    for key, value in incoming_edition_dict.items():
+        # Skip fields that shouldn't end up in an edition:
+        if key == "cover":
```

Comment:
this shouldn't be needed if the cover had already been correctly substituted in:
 
https://github.com/internetarchive/openlibrary/blob/3ef1957dbb4d30f66892a6c3a55c1f812b64a4ce/openlibrary/catalog/add_book/__init__.py#L851-L857

I think this check is happening too early, but then I don't think it should be a special case either.

Reply:
I removed this whole section pursuant to your suggestion about creating a more general solution.